### ISSUE TITLE
Deal with inconsistent optimization

### ISSUE THREAD

**Issue by kxz18** [kxz18/boltz_bindcraft#4]
The loss curves vibrate a lot. There are several possible reasons:

1. the discretization breaks the consistency -> maintain_logits
2. the different seed in the inner loop causes inconsistent gradient -> reset the seed before each step

<img width="1125" height="850" alt="Image" src="https://github.com/user-attachments/assets/bf3fa61b-cb5a-4627-94a8-cefd55afd867" />

**Comment by kxz18** [comment]
Better to use less diffusion steps, otherwise the accumulating numerical error will make the gradient inaccurate.
